nest_elephant_tvb/translation/mpi_translator.py

```diff
@@ -3,192 +3,237 @@
+            --> For now, hardcoded solution. All places with 'rank 0' are replaced with 'rank 1'
+            --> But as soon as we use more than two MPI ranks, this fails again!!!
     
+    TODO: Renaming! Translator -> transformer, etc.
+    TODO: This is only the Nest to TVB direction. For other direction some generalizing changes will be needed.
+    TODO: Use RichEndPoints for communication encapsulation
+    TODO: Seperate 1)Receive 2)Analysis/Science and 3)Send. See also the many Todos in the code
+    TODO: Make use of / enable MPI parallelism! Solve hardcoded communication protocol first
+    TODO: Major question and idea: 
+            Why all the trouble with 'two parts' if everything is synchronized?
+            We receive and send seperately but still wait for each other.
+            With MPI possible --> ONE while-loop that handles sending and receiving.
+            Send/Recv seperated by ranks obviously (paralellism).
     '''
     
     # destructure logger list to indivual variables
     logger_master, logger_receive, logger_send = loggers
     # science part, see import
     # TODO: use os.path (or similar) for proper file handling.
+    # TODO: move this object creation to a proper place. They are passed through many functions.
     store = store_data(path+'/log/',param)
     analyse = analyse_data(path+'/log/',param)
-    # variable for communication between thread
-    status_data=[0] # status of the buffer
     
-    ############ NEW Code: MPI intracommunicator
-    # receive on rank 0, science and send on rank 1
+    ############ NEW Code: 
     # TODO: future work: mpi parallel, use rank 1-x for science and sending
-    # NOTE: still copy-paste from kim master thesis code....
-    # new server-intracommunicator without receiving rank 0
-    intracomm = comm.Create(comm.Get_group().Excl([0]))
+    # TODO: use this MPI intracommunicator, without receiving rank 0
+    # intracomm = comm.Create(comm.Get_group().Excl([0]))
     # create the shared memory block / databuffer
-    buf_pkgs = 100 # NOTE: size hardcoded!
-    databuffer = _shared_mem_buffer(buf_pkgs,100,comm)
+    databuffer = _shared_mem_buffer(comm)
     ############# NEW Code end
     
-    # TODO:
-    # 1) receive package from nest, store in shared mem buffer
-    # 2) take from shared mem buffer, do science, send to tvb
-    # solve issues:
-    # - one package incoming, one package outgoing (synchronize)
-    # - mark buffer if new package arrived --> not to slow not to fast 
-    # - FIRST idea: 
-    #   - Receiving rank 0 check for 'NaN' in buffer, i.e. is it empty?
-    #   - if NaN then fill with new package
-    #   - Sending rank 1 (or more in the future) check for new package in buffer (not NaN)
-    #   - do science and send to TVB
-    #   - set buffer to NaN to mark as ready
-    # TODO: how to check for END of simulation?
+    ############ NEW Code: Receive/analyse/send
+    if comm.Get_rank() == 0: # Receiver from NEST, rank 0
```

Comment:
Add a clear mark that this is due to the other side implementation and is a TODO

Reply:
Added a TODO and a bit more explanation.